### PR TITLE
Add boardsource5x12

### PR DESCRIPTION
**Pull request by fsargent**
## Board/Shield Check-list
 - [X] This board/shield is tested working on real hardware
 - [X] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [X] `.zmk.yml` metadata file added
 - [X] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [X] General consistent formatting of DeviceTree files
 - [X] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [X] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [X] If split, no name added for the right/peripheral half
 - [X] `.conf` file has optional extra features commented out

### CODE REVIEW

File: app/boards/shields/boardsource5x12/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/Kconfig.defconfig:1]: Should be 2021:

```suggestion
# Copyright (c) 2021 The ZMK Contributors
```

File: app/boards/shields/boardsource5x12/Kconfig.shield [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/Kconfig.shield:1]:
```suggestion
# Copyright (c) 2021 The ZMK Contributors
```

File: app/boards/shields/boardsource5x12/boardsource5x12.keymap [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/boardsource5x12.keymap:2]:
```suggestion
 * Copyright (c) 2021 The ZMK Contributors
```

File: app/boards/shields/boardsource5x12/boardsource5x12.overlay [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/boardsource5x12.overlay:2]:
```suggestion
 * Copyright (c) 2021 The ZMK Contributors
```

**Review by petejohanson — CHANGES_REQUESTED** [review]
A few minor things.

**Comment by fsargent** [comment]
Done

File: app/boards/shields/boardsource5x12/boardsource5x12.keymap [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            bindings = <
+                &kp GRAV    &kp N1      &kp N2   &kp N3    &kp N4    &kp N5      &kp N6    &kp N7    &kp N8    &kp N9   &kp N0   &kp BSPC
+                &kp TAB     &kp Q       &kp W    &kp E     &kp R     &kp T       &kp Y     &kp U     &kp I     &kp O    &kp P    &kp DEL
+                &kp ESC     &kp A       &kp S    &kp D     &kp F     &kp G       &kp H     &kp J     &kp K     &kp L    &kp SEMI &kp SQT
+                &kp LSHIFT  &kp Z       &kp X    &kp C     &kp V     &kp B       &kp N     &kp M     &kp COMMA &kp DOT  &kp FSLH &kp RET
+                &kp PIPE    &kp RCTRL   &kp LALT &kp LGUI  &kp LOWER &kp SPACE   &kp SPACE &kp RAISE &kp LEFT  &kp DOWN &kp UP   &mo RIGHT
+                >;
+            };        
+            
+        RAISE {
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/boardsource5x12.keymap:29]: These should be lowercase and using the convention:

```suggestion
        raise_layer {
```

File: app/boards/shields/boardsource5x12/boardsource5x12.keymap [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            bindings = <
+                &kp GRAV    &kp F1      &kp F2   &kp F3    &kp F4    &kp F5      &kp F6    &kp F7    &kp F8    &kp F9       &kp F10         &kp F11
+                &kp ESC     &kp N1      &kp N2   &kp N3    &kp N4    &kp N5      &kp N6    &kp N7    &kp N8    &kp N9       &kp N0          &kp F12
+                &kp ESC     &kp N4      &kp N5   &kp N6    &kp PLUS  &kp G       &kp H     &kp J     &kp K     &kp L        &kp SEMI        &kp SQT
+                &kp RET     &kp N7      &kp N8   &kp N9    &kp MINUS &kp B       &kp N     &kp M     &kp COMMA &kp DOT      &kp FSLH        &kp RET
+                &reset      &bootloader &kp LALT &kp LGUI  &kp LOWER &kp SPACE   &kp SPACE &kp RAISE &kp LEFT  &kp C_VOL_DN &kp C_VOL_UP    &mo C_PLAY_PAUSE
+                >;
+            };
+        LOWER {
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/boardsource5x12.keymap:38]:
```suggestion
        lower_layer {
```

File: app/boards/shields/boardsource5x12/boardsource5x12.keymap [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            bindings = <
+                &kp ESC    &kp F1       &kp F2   &kp F3    &kp F4    &kp F5      &kp F6    &kp F7    &kp F8    &kp F9   &kp F10  &kp F11
```
Comment (petejohanson) [inline app/boards/shields/boardsource5x12/boardsource5x12.keymap:40]: We usually aim to have the `&bt` behaviors bound somewhere in the default keymap somewhere too.

**Review by petejohanson — CHANGES_REQUESTED** [review]
A few more items I missed on the first pass. Thanks!

**Comment by fsargent** [comment]
Ready for re-review

File: app/boards/shields/boardsource5x12/boardsource5x12.keymap [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            bindings = <
+                &kp ESC    &kp N1       &kp N2      &kp N3   &kp N4     &kp N5      &kp N6      &kp N7  &kp N8      &kp N9      &kp N0      &kp BSPC
+                &kp TAB    &kp Q        &kp W       &kp E    &kp R      &kp T       &kp Y       &kp U   &kp I       &kp O       &kp P       &kp BSLH
+                &kp LSHFT  &kp A        &kp S       &kp D    &kp F      &kp G       &kp H       &kp J   &kp K       &kp L       &kp SEMI    &kp SQT
+                &kp LCTRL  &kp Z        &kp X       &kp C    &kp V      &kp B       &kp N       &kp M   &kp COMMA   &kp DOT     &kp FSLH    &kp RET
+                &mo 3      &kp LCTRL    &kp LALT    &kp LGUI &mo 1      &kp SPACE   &kp SPACE   &mo 2   &kp LEFT    &kp DOWN    &kp UP      &kp RIGHT 
+            >;
+        };
+
+        lower {
```
Comment (Nicell) [inline app/boards/shields/boardsource5x12/boardsource5x12.keymap:35]:
```suggestion
        lower_layer {
```
These layers should follow suit with the `default_layer`

**Review by Nicell — COMMENTED** [review]
Just one change left from what I can see 😄

**Comment by fsargent** [comment]
Resolved @Nicell

**Comment by Nicell** [comment]
Hey @fsargent, there's still the rest of the layers after `lower_layer` that needs `_layer` added 😄

**Comment by fsargent** [comment]
Sorry, done!

**Review by Nicell — APPROVED** [review]

**Comment by fsargent** [comment]
🎉

**Pull request merged**